CODEOWNERS.md

```diff
@@ -0,0 +1,2 @@
+# PRs require approval from the earthly core team
```

Comment:
```suggestion
# PRs require approval from the earthly engineering team
```

or maybe we should just delete a comment like this since having the line below could be sufficient (and won't allow a comment to become out of date)